mne/encoding/model.py

```diff
@@ -0,0 +1,238 @@
+from ..externals.six import string_types
+
+
+class SampleMasker(object):
+    """Remove samples with a particular value before fitting.
+
+    Parameters
+    ----------
+    estimator : instance of sklearn-style estimator
+        The estimator to be called after samples are removed.
+    mask_val : float, int, nan | callable | None
+        The value that will be removed from the dataset. If a callable,
+        it should return a boolean mask of length n_samples that takes
+        X an array of shape (n_samples, n_features) as input.
+    samples_train : array, shape (n_samples,) | None
+        An optional array of indices for manually specifying which samples
+        to keep during training. Defaults to None.
+    samples_pred : array, shape (n_samples,) | None
+        An optional array of indices for manually specifying which samples
+        to keep during prediction. Defaults to None.
+    mask_condition : 'all', 'any'
+        Whether only 1 `mask_val` will trigger removal of a row, or if
+        all values per row must be `mask_val` for removal. Defaults to 'all'.
+    """
+    def __init__(self, estimator, mask_val=None, samples_train=None,
+                 samples_pred=None, mask_condition='all'):
+        if all(ii is not None for ii in [mask_val, samples_train]):
+            raise ValueError('Supply one of mask_val | samples_train, or both')
+        if samples_train is not None:
+            samples_train = np.asarray(samples_train).astype(int)
+        self.samples_train = samples_train
+        if samples_pred is not None:
+            samples_pred = np.asarray(samples_pred).astype(int)
+        self.samples_pred = samples_pred
+
+        self.mask_val = mask_val
+        if mask_condition not in ['all', 'any']:
+            raise ValueError('condition must be one of "all" or "any"')
+        self.mask_condition = mask_condition
+        self.est = _check_estimator(estimator)
+
+    def fit(self, X, y):
+        """Remove datapoints then fit the estimator.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Samples (rows) will be
+            removed prior to fit according to `mask_val` and `samples_train`.
+        y : array, shape (n_samples, n_targets)
+            The output array for the model. Samples (rows) will be
+            removed prior to fit according to `mask_val` and `samples_train`.
+        """
+        X, y = self._mask_data(X, y, ixs=self.samples_train)
+        self.est.fit(X, y)
+        return self
+
+    def predict(self, X):
+        """Call the estimator within `self`.
+
+        This will call `self.est.predict`.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            The input array for the model. Samples (rows) will be
+            removed prior to fit according to `samples_pred`.
+        """
+        X = self._mask_data(X, ixs=self.samples_pred)
```

Comment:
It's already doing something like this with `ixs` (if the object is initialized with `samples_pred=None` then it will pass `None` to this function on the call to `predict`, and no subsampling will occur). I could add a `mask_vals_pred` parameter that could be `None | 'same' | an arbitrary list`